test/promise.test.satyg

```diff
@@ -0,0 +1,20 @@
+@require: test/test
+@require: test/expect
+
+@import: ../src/base0
+@import: ../src/ref
+@import: ../src/promise
+
+let promise-test-cases = open Test in
+  describe `promise module` [
+    it `evaluate lazily` (fun () -> (
```

Comment:
1. The original test uses `Debug.log` to check the order of side effects, but this is not feasible with the satisfy-test. Instead, this test case uses `ref` to make side effects.
2. The original test has additional cases that I did not find helpful. That's why they are omitted in this test case.